### ISSUE TITLE
Typescript cannot find type declaration file when using import

### ISSUE THREAD

**Issue by everett1992** [iter-tools/iter-tools#490]
When using typescript's node16 module and 'module' entry points typescript cannot find iter-tools's type definitions. 

### Setup
package.json
```
{
  "dependencies": {
    "@types/node": "^18.11.18",
    "iter-tools": "^7.5.0",
    "typescript": "^4.9.4"
  }
}
```
tsconfig.json
```
{
  "compilerOptions": {
    "strict": true,
    "target": "ES2022",
    "module": "node16",
    "moduleResolution": "node16",
  }
}
```
index.cts
```
import {asyncMap} from 'iter-tools'
```
index.mts
```
import {asyncMap} from 'iter-tools'
```

```
$ npx tsc --noEmit
index.mts:1:24 - error TS7016: Could not find a declaration file for module 'iter-tools'. 'test/node_modules/iter-tools/index.mjs' implicitly has an 'any' type.
  If the 'iter-tools' package actually exposes this module, try adding a new declaration (.d.ts) file containing `declare module 'iter-tools';`

1 import {asyncMap} from 'iter-tools'
```

**Comment by everett1992** [comment]
It can be partially fixed by changing the `exports` in package.json but I couldn't figure out how to fix the `./methods/*` export
```diff
 "exports": {
   ".": {
   "import": "./index.mjs",
   "require": "./index.js"
+  "types": "./index.d.ts"
   }
 }
 ```

**Comment by conartist6** [comment]
What do you mean partially fixed?

**Comment by everett1992** [comment]
It fixes `import {} from 'iter-tools'` but not `import {} from 'iter-tools/methods/batch`. I thought that adding `"types": "./methods/*.d.ts" did not work but I checked again and it does.

I'd open a pull request but I don't understand the difference between package.json, package.es.json, and package.es5.json.

**Comment by conartist6** [comment]
Oh yeah that's a bit tricky. `package.json` is used when you install the github repo as a package, i.e. `npm i iter-tools-es@github:iter-tools/iter-tools`.

Packages tarballs that go to npm are built with `scripts/dist.cjs`:
- `iter-tools/package.json` is built by merging `package.json` with `package.es5.json`
- `iter-tools-es/package.json` is built by merging `package.json` with `package.es.json`

The main difference when using the github repo as a package is that the `exports` must be prefixed with `./src`, while neither of the published packages need that prefix (hence why both override files define different `exports`). **I think that means the fix will need to touch all three package files.**

**Comment by bodograumann** [comment]
What do you think about my proposal, @conartist6 ?

**Comment by conartist6** [comment]
Fixed in `iter-tools@7.5.2`
